### PR TITLE
fix: remove api keys from RPC error messages

### PR DESCRIPTION
**Pull request by agostbiro**
Only log the host instead of the full URL in RPC errors to avoid leaking api keys in CI.

### CODE REVIEW

File: crates/edr_eth/src/remote/client.rs [hunk 1]
```diff
@@ -1203,7 +1206,7 @@ mod tests {
 
         if let RpcClientError::HttpStatus(error) = error {
             assert_eq!(
-                error.status(),
+                <ReqwestError as Into<reqwest::Error>>::into(error).status(),
```
Comment (Wodann) [inline crates/edr_eth/src/remote/client.rs:1209]: Why did `reqwest::Error::from(..)` not work here?

File: crates/edr_eth/src/remote/client.rs [hunk 2]
```diff
@@ -1245,20 +1248,26 @@ mod tests {
...
                 .call::<Option<eth::Transaction>>(RequestMethod::GetTransactionByHash(hash))
                 .await
                 .expect_err("should have failed to interpret response as a Transaction");
 
+            assert!(!error.to_string().contains(api_key));
+
             if let RpcClientError::HttpStatus(error) = error {
-                assert_eq!(error.status(), Some(StatusCode::from_u16(401).unwrap()));
+                assert_eq!(
+                    <ReqwestError as Into<reqwest::Error>>::into(error).status(),
```
Comment (Wodann) [inline crates/edr_eth/src/remote/client.rs:1268]: Same question w.r.t `From::from`

File: .changeset/brave-scissors-train.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+
```
[the remark this answers is not in the document]
Comment (agostbiro) [reply .changeset/brave-scissors-train.md]: thanks!

File: crates/edr_eth/src/remote/client.rs (hunk 1 above)
Comment (agostbiro) [reply crates/edr_eth/src/remote/client.rs]: Yeah that's cleaner thanks, fixed in [Simplify error conversion](https://github.com/NomicFoundation/hardhat/pull/5067/commits/fa0abb8e7574ecdeae2c4a4a6d6761d27ac52023)

File: crates/edr_eth/src/remote/client.rs (hunk 2 above)
Comment (agostbiro) [reply crates/edr_eth/src/remote/client.rs]: fixed in [Simplify error conversion](https://github.com/NomicFoundation/hardhat/pull/5067/commits/fa0abb8e7574ecdeae2c4a4a6d6761d27ac52023)

**Review by fvictorio — APPROVED** [review]

**Pull request merged**